### PR TITLE
refactor: [M3-8045] - remove aria-label from TableRow

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
While aria-label can be used on any element that can have an accessible name, in practice however, it is supported only on interactive elements, [widgets](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles), [landmarks](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#3._landmark_roles), images, and iframes.

While aria-label can be used on any element that has an accessible name, it's best supported for interactive elements, widgets, landmarks, images, and iframes. In general, it shouldn't be used on static content. Using aria-label on non-interactive elements **can cause a confusing user experience or the label may not be announced by assistive technology.** For example, Microsoft's screen reader, JAWS, ignores aria-label on all static content except for content with “an interactive role, img role, or nav, main, search roles”. 

also: https://stackoverflow.com/questions/72568204/what-can-aria-label-be-used-on

In the case of our table rows, having an aria label dos not really make sense:
- Our tables already have accessible elements, which is encouraged albeit not being technically interactive 
> `aria-label`, `aria-labelledby` and `aria-describedby` work well on `table`, `th` and `td` elements with a few exceptions for NVDA, VoiceOver on iOS, and Talkback discussed in next section. 
see this [link](https://www.w3.org/TR/using-aria/#practical-support-aria-label-aria-labelledby-and-aria-describedby)
- all we were doing (which this PR removes) is taking the label in the first column and use it as an aria label on the row. The table context provides enough accessibility with the table header and its content.

## Changes  🔄
- Remove `ariaLabel` on `<TableRow />`
- Update components passing the prop

## Preview 📷
No visual changes should be expected as part of this PR

### Verification steps
- There isn't much to verify here, except that our tests pass and the tag was properly removed from the rows affected by this PR

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

**Review by hana-linode — COMMENTED** [review]
Doing a search for `ariaLabel` in the codebase, would the `StatusIcon` and `StyledStatusIcon` components be a good contender for removal as well since they aren't interactive?

**Review by dwiley-akamai — APPROVED** [review]
Code review ✅
Tests pass ✅

File: packages/manager/src/components/CollapsibleTable/CollapsibleTable.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ export const CollapsibleTable = (props: Props) => {
   return (
     <TableContainer component={Paper}>
       <Table aria-label="collapsible table">
-        <TableHead>{TableRowHead}</TableHead>
+        <TableHead data-qa-table-row="vpc-subnets">{TableRowHead}</TableHead>
```
[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/src/components/CollapsibleTable/CollapsibleTable.tsx:30]: oh yes good point

[the remark this answers is not in the document]
Comment (abailly-akamai) [reply packages/manager/src/components/CollapsibleTable/CollapsibleTable.tsx]: done

**Comment by abailly-akamai** [comment]
@hana-linode yup! adding those to my epic, thanks for pointing it out

**Review by hana-linode — APPROVED** [review]

**Pull request merged**